src/overlays/actors/ovl_Obj_Mure/z_obj_mure.c

```diff
@@ -26,43 +39,387 @@ const ActorInit Obj_Mure_InitVars = {
+                     thisx->params);
+    }
+}
 
-#pragma GLOBAL_ASM("asm/non_matchings/overlays/actors/ovl_Obj_Mure/func_80B98D18.s")
+void ObjMure_Destroy(Actor* thisx, GlobalContext* globalCtx) {
+}
 
-#pragma GLOBAL_ASM("asm/non_matchings/overlays/actors/ovl_Obj_Mure/func_80B98F38.s")
+s32 ObjMure_GetMaxChildSpawns(ObjMure* this) {
+    if (this->chNum == 0) {
+        return sMaxChildSpawns[this->ptn];
+    }
+    return this->chNum;
+}
 
-#pragma GLOBAL_ASM("asm/non_matchings/overlays/actors/ovl_Obj_Mure/func_80B990BC.s")
+void ObjMure_GetSpawnPos(Vec3f* outPos, Vec3f* inPos, s32 ptn, s32 idx) {
+    if (ptn >= 4) {
+        osSyncPrintf("おかしなの (%s %d)\n", "../z_obj_mure.c", 307);
+    }
+    *outPos = *inPos;
+}
 
-#pragma GLOBAL_ASM("asm/non_matchings/overlays/actors/ovl_Obj_Mure/func_80B9910C.s")
+void ObjMure_SpawnActors0(ObjMure* this, GlobalContext* globalCtx) {
+    ActorContext* ac;
+    s32 i;
+    Vec3f pos;
+    s32 pad;
+    s32 maxChildren = ObjMure_GetMaxChildSpawns(this);
 
-#pragma GLOBAL_ASM("asm/non_matchings/overlays/actors/ovl_Obj_Mure/func_80B99204.s")
+    for (i = 0; i < maxChildren; i++) {
+        if (this->children[i] != NULL) {
+            // Translation: "Error: I already have a child(%s %d)(arg_data 0x%04x)
+            osSyncPrintf("Error : 既に子供がいる(%s %d)(arg_data 0x%04x)\n", "../z_obj_mure.c", 333,
+                         this->actor.params);
+        }
+        switch (this->childrenStates[i]) {
+            case OBJMURE_CHILD_STATE_1:
+                break;
+            case OBJMURE_CHILD_STATE_2:
+                ac = &globalCtx->actorCtx;
+                ObjMure_GetSpawnPos(&pos, &this->actor.world.pos, this->ptn, i);
+                this->children[i] =
+                    Actor_Spawn(ac, globalCtx, sSpawnActorIds[this->type], pos.x, pos.y, pos.z, this->actor.world.rot.x,
+                                this->actor.world.rot.y, this->actor.world.rot.z, sSpawnParams[this->type]);
+                if (this->children[i] != NULL) {
+                    this->children[i]->flags |= 0x800;
+                    this->children[i]->room = this->actor.room;
+                } else {
+                    osSyncPrintf("warning 発生失敗 (%s %d)\n", "../z_obj_mure.c", 359);
+                }
+                break;
+            default:
+                ac = &globalCtx->actorCtx;
+                ObjMure_GetSpawnPos(&pos, &this->actor.world.pos, this->ptn, i);
+                this->children[i] =
+                    Actor_Spawn(ac, globalCtx, sSpawnActorIds[this->type], pos.x, pos.y, pos.z, this->actor.world.rot.x,
+                                this->actor.world.rot.y, this->actor.world.rot.z, sSpawnParams[this->type]);
+                if (this->children[i] != NULL) {
+                    this->children[i]->room = this->actor.room;
+                } else {
+                    osSyncPrintf("warning 発生失敗 (%s %d)\n", "../z_obj_mure.c", 382);
+                }
```

Comment:
I suggest adding a break here if possible
```suggestion
                }
                break;
```